dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -371,45 +373,47 @@ Path findClassLocation(String className) {
    * Returns mapping from class names to the absolute paths of the first jar file that defines the
    * classes.
    *
-   * @param jarFilePaths absolute paths to jar files
+   * @param jars absolute paths to jar files
    */
   @VisibleForTesting
-  static ImmutableMap<String, Path> mapClassToJar(List<Path> jarFilePaths)
-      throws IOException {
+  static ImmutableMap<String, Path> mapClassToJar(List<Path> jars) throws IOException {
     Map<String, Path> classToJar = new HashMap<>();
-    for (Path jarFilePath : jarFilePaths) {
-      for (JavaClass javaClass : listClassesInJar(jarFilePath)) {
+    for (Path jar : jars) {
+      for (String className : listClassNamesInJar(jar)) {
         // The first entry wins in the same manner as JVM's class loading.
-        classToJar.putIfAbsent(javaClass.getClassName(), jarFilePath);
+        classToJar.putIfAbsent(className, jar);
       }
     }
     return ImmutableMap.copyOf(classToJar);
   }
 
-  private static ImmutableSet<ClassInfo> listClassInfo(URL jarUrl)
-      throws IOException {
+  static ImmutableSet<String> listClassNamesInJar(Path jar) throws IOException {
+    URL jarUrl = jar.toUri().toURL();
     // Setting parent as null because we don't want other classes than this jar file
-    URLClassLoader classLoaderFromJar = new URLClassLoader(new URL[]{jarUrl}, null);
+    URLClassLoader classLoaderFromJar = new URLClassLoader(new URL[] {jarUrl}, null);
 
     // Leveraging Google Guava reflection as BCEL doesn't list classes in a jar file
     com.google.common.reflect.ClassPath classPath =
         com.google.common.reflect.ClassPath.from(classLoaderFromJar);
 
-    return classPath.getAllClasses();
+    return classPath.getAllClasses().stream()
+        .map(ClassInfo::getName)
+        .collect(toImmutableSet());
   }
 
-  static ImmutableSet<JavaClass> listClassesInJar(Path jarPath) throws IOException {
-    Repository repository = createClassRepository(ImmutableList.of(jarPath));
```

Comment:
This (creating a `Repository` with a new class path that has only this jar file) was the cause of the bug. When listing `JavaClass`, it was not using the class path passed to `ClassDumper` constructor.